internal/resource/config/oauth/oauth_auth_server_settings_data_source.go

```diff
@@ -0,0 +1,620 @@
+				Computed:    true,
+			},
+			"approved_authorization_detail_attribute": schema.StringAttribute{
+				Description: "Attribute from the external consent adapter's contract, intended for storing approved authorization details returned by the external consent page.",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"par_reference_timeout": schema.Int64Attribute{
+				Description: "The timeout, in seconds, of the pushed authorization request reference. The default value is 60.",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"par_reference_length": schema.Int64Attribute{
+				Description: "The entropy of pushed authorization request references, in bytes. The default value is 24.",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"par_status": schema.StringAttribute{
+				Description: "The status of pushed authorization request support. The default value is ENABLED.",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{"DISABLED", "ENABLED", "REQUIRED"}...),
+				},
+			},
+			"client_secret_retention_period": schema.Int64Attribute{
+				Description: "The length of time in minutes that client secrets will be retained as secondary secrets after secret change. The default value is 0, which will disable secondary client secret retention.",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"jwt_secured_authorization_response_mode_lifetime": schema.Int64Attribute{
+				Description: "The lifetime, in seconds, of the JWT Secured authorization response. The default value is 600.",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+		},
+	}
+	id.ToDataSourceSchema(&schemaDef, false, "The ID of this resource.")
+	resp.Schema = schemaDef
+}
+
+// Metadata returns the data source type name.
+func (r *oauthAuthServerSettingsDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_oauth_auth_server_settings"
+}
+
+// Configure adds the provider configured client to the data source.
+func (r *oauthAuthServerSettingsDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+}
+
+// Read a OauthAuthServerSettingsResponse object into the model struct
+func readOauthAuthServerSettingsResponseDataSource(ctx context.Context, r *client.AuthorizationServerSettings, state *oauthAuthServerSettingsDataSourceModel, existingId *string) diag.Diagnostics {
+	var diags, respDiags diag.Diagnostics
+	state.Id = id.GenerateUUIDToState(existingId)
```

Comment:
I'm wondering if there might need to be a UseStateForUnknown on the data source id attribute when we are generating a UUID. Could you check if the resource id changes on subsequent apply? We would want it to randomly generate an id on the first apply and then just maintain that value.

Reply:
/How would I do this without using a planmodifier?

Reply:
Ah, yeah I forgot that the data source schema doesn't include plan modifiers.

It looks like you are already getting the state in your read method with req.Config.Get(). You can just read the value from the state struct after that call.